rbac_api.go

```diff
@@ -55,6 +55,22 @@ func (e *Enforcer) AddRoleForUser(user string, role string) (bool, error) {
 	return e.AddGroupingPolicy(user, role)
 }
 
+// AddRolesForUser adds roles for a user.
+// Returns false if the user already has the roles (aka not affected).
+func (e *Enforcer) AddRolesForUser(user string, roles []string) (bool, error) {
+    f := false
+    for _, r := range roles {
```

Comment:
Calling `AddGroupingPolicies` is better.

Reply:
How does `AddGroupingPolicies` work? what is the structure of `rules` in this case?

Reply:
See `management_api_test.go` for `AddGroupingPolicies` test cases.

Reply:
There might be a bug in `AddGroupingPolicies` because using the code below makes the user only has one role so it looks like the policies are being overwritten somehow.

```
func (e *Enforcer) AddRolesForUser(user string, roles []string) (bool, error) {
	var rules [][]string
	for _, r := range roles {
		rules = append(rules, []string{user, r})
	}
	return e.AddGroupingPolicies(rules)
}
```

Reply:
Casbin supports a user to inherit multiple roles.

Reply:
Though `AddGroupingPolicies` does not allow adding multiple roles to the same user.

Reply:
> Though `AddGroupingPolicies` does not allow adding multiple roles to the same user.

Could you point out where it is?

Reply:
if you use the above code on my branch you would get an error:
```
=== RUN   TestEnforcer_AddRolesForUser
2020/05/10 16:10:20 Role links for: g
2020/05/10 16:10:20 alice < data2_admin
    TestEnforcer_AddRolesForUser: rbac_api_test.go:133: Roles for  alice :  [data2_admin]
    TestEnforcer_AddRolesForUser: rbac_api_test.go:133: Roles for  alice :  [data2_admin] , supposed to be  [data1_admin data2_admin data3_admin]
```

only one role ended up added to the user which is `data2_admin`

Reply:
You ignore the returns of `e.AddRolesForUser`. `e.AddRolesForUser("alice", []string{"data1_admin", "data2_admin", "data3_admin"})` returns `false, nil`, so you get `[data2_admin]`.

`AddGroupingPolicies ` has  atomic operation. Due to `alice` has `data2_admin` role, so cannot add `data2_admin` to `alice` and returns `false, nil`.

Reply:
Then using `AddGroupingPolicy` is a better  option because we need to make those the roles are added even though one of them was already added before.

Reply:
> Then using `AddGroupingPolicy` is a better option because we need to make those the roles are added even though one of them was already added before.

Sounds good. 

What do you think?  @hsluoyz @GopherJ @DivyPatel9881

Reply:
> What do you think? @hsluoyz @GopherJ @DivyPatel9881

I agree.